### PR TITLE
Added the overridable method getSharedPreferences() 

### PR DESCRIPTION
**Pull request by Yubyf**
To provide a custom implementation interface for SharedPreferences(like [DataStore](https://developer.android.google.cn/reference/kotlin/androidx/datastore/core/DataStore)).

### CODE REVIEW

**Comment by apsun** [comment]
Implementation looks good, although I'm unclear how you will actually use this. DataStore doesn't seem to implement the SharedPreferences interface as far as I can tell (unless you plan to write some kind of DataStore -> SharedPreferences adapter?)

**Comment by Yubyf** [comment]
> Implementation looks good, although I'm unclear how you will actually use this. DataStore doesn't seem to implement the SharedPreferences interface as far as I can tell (unless you plan to write some kind of DataStore -> SharedPreferences adapter?)

Yes, some time ago when I migrated SharedPreferences to DataStore, I found that I still needed to adapt the interface of SharedPreferences in some places (like RemotePreferences).

Therefore, I wrote a simple implementation [here](https://github.com/Yubyf/DataStorePreferences).

Since this library is not elegant enough and needs to use Kotlin Coroutine, I did not commit it here.

File: library/src/main/java/com/crossbowffs/remotepreferences/RemotePreferenceProvider.java [hunk 1]
```diff
@@ -143,13 +143,21 @@ public boolean onCreate() {
...
             mPreferences.put(file.getFileName(), prefs);
         }
         return true;
     }
 
+    /**
+     * Generate {@link SharedPreferences} to store the key-value data.
+     * Override this method to provide a custom implementation of {@link SharedPreferences}.
+     */
+    protected SharedPreferences getSharedPreferences(Context context, String name) {
```
Comment (apsun) [inline library/src/main/java/com/crossbowffs/remotepreferences/RemotePreferenceProvider.java]:
```suggestion
    protected SharedPreferences getSharedPreferences(Context context, String prefFileName) {
```

For consistency with the rest of the code and to make it a bit more obvious what 'name' refers to

File: library/src/main/java/com/crossbowffs/remotepreferences/RemotePreferenceProvider.java [hunk 2]
```diff
@@ -143,13 +143,21 @@ public boolean onCreate() {
...
+            SharedPreferences prefs = getSharedPreferences(context, file.getFileName());
             prefs.registerOnSharedPreferenceChangeListener(this);
             mPreferences.put(file.getFileName(), prefs);
         }
         return true;
     }
 
+    /**
+     * Generate {@link SharedPreferences} to store the key-value data.
+     * Override this method to provide a custom implementation of {@link SharedPreferences}.
```
Comment (apsun) [inline library/src/main/java/com/crossbowffs/remotepreferences/RemotePreferenceProvider.java:155]:
```suggestion
     * Generate {@link SharedPreferences} to store the key-value data.
     * Override this method to provide a custom implementation of {@link SharedPreferences}.
     *
     * @param context The context that should be used to get the preferences object.
     * @param prefFileName The name of the preference file.
     * @return An object implementing the {@link SharedPreferences} interface.
```

Mind adding javadoc for the parameters? Normally I hate writing obvious docs but the rest of the file does it in this style :p

**Review by apsun — CHANGES_REQUESTED** [review]

**Comment by apsun** [comment]
I see, makes sense! This might also be useful if someone ever wants to chain multiple RemotePreference instances together, I guess. I just have some minor nits regarding naming and docs, otherwise this LGTM.

File: library/src/main/java/com/crossbowffs/remotepreferences/RemotePreferenceProvider.java (hunk 2 above)
Comment (Yubyf) [reply library/src/main/java/com/crossbowffs/remotepreferences/RemotePreferenceProvider.java:155]: As you wish.

**Review by Yubyf — COMMENTED** [review]
Accepted. I took your suggestion and overwrote the old commit by forcing a new one to be pushed to keep the branch clean.

**Pull request merged**